### ISSUE TITLE
REST Client is added to OpenAPI file

### ISSUE THREAD

**Issue by andlinger** [quarkusio/quarkus#24344]
### Describe the bug

When I use a REST Reactive client in my Quarkus project, which has a path annotation at the class level, it is automatically added to the OpenAPI specification. Unfortunately hiding it via `@Schema(hidden = true)` does not work.

Following client shows in OpenAPI specification:
```java
@RegisterRestClient
@Path("/get")
public interface VisibleHttpClient {

    @GET
    String get();
}
```

Following client isn't shown:
```java 
@RegisterRestClient
public interface InvisibleHttpClient {

    @GET
    @Path("/get")
    String get();
}
```

### Expected behavior

REST clients shouldn't be added to OpenAPI spec.

### How to Reproduce?

https://github.com/andlinger/quarkus-reproducer/tree/quarkus-24344

### Quarkus version or git rev

2.7.4

**Comment by gsmet** [comment]
@EricWittmann, @MikeEdgar, @michalszynkiewicz, @phillip-kruger please have a look at this one for 2.8 as it's part of our RESTEasy Reactive story.

**Comment by MikeEdgar** [comment]
@andlinger do you have a reproducer? I'm guessing the scanner is including it due to some generated implementation of the interface being generated that is different than what we currently consider "synthetic".

**Comment by andlinger** [comment]
@MikeEdgar The reproducer is already mentioned in the issue

**Comment by gsmet** [comment]
It would probably be interesting to compare with what was done for RESTEasy Classic.

**Comment by andlinger** [comment]
@MikeEdgar No problem :) I also suspect that the problem is in the generated code. In the OpenAPI specification the interface is called "CDI Wrapper".

**Comment by michalszynkiewicz** [comment]
@MikeEdgar are you looking into this issue?

**Comment by MikeEdgar** [comment]
@michalszynkiewicz I haven't started anything yet. I was planning on taking a look in my evening or in the morning unless you're interested in grabbing it.

**Comment by michalszynkiewicz** [comment]
if you have time for it, I'd be grateful if you gave it a try. Ping me on Zulip if you need help.

**Comment by MikeEdgar** [comment]
- Class `VisibleHttpClient` has flags `0x601` (0x400 abstract/0x200 interface)
- Generated class `VisibleHttpClient$$CDIWrapper` has flags `0x1021` (0x1000 synthetic)
- Class `InvisibleHttpClient` is skipped because it is not annotated with `@Path`

smallrye-open-api already has a check to exclude synthetic interfaces. Has there been a change to how these classes are generated/represented in the past few months? We can add an exclusion for synthetic classes as well.

**Comment by MikeEdgar** [comment]
So it seems the difference is that in the earlier Quarkus version, the `VisibleHttpClient$$CDIWrapper` class did not have the `@Path` annotation present, but it is present in 2.7.

Fixed with https://github.com/smallrye/smallrye-open-api/pull/1096

**Comment by gsmet** [comment]
@MikeEdgar is it a 2.7 related change or the fact that the project is using the **Reactive** REST Client?

It's important because if it's a regression in 2.7, we might need a backport of the fix to the 2.7 branch.

**Comment by MikeEdgar** [comment]
@gsmet the original support we added to ignore synthetic implementations (https://github.com/smallrye/smallrye-open-api/pull/919) was also there for the Reactive REST client, so that much is in common. This looks like a change to how the Reactive client generation works between 2.2.5 and 2.7.5. It doesn't necessarily seem like a regression from that perspective, but rather a case the OpenAPI scanner didn't expect.

**Comment by gsmet** [comment]
Well it's sort of a regression if it worked well before and it now pollutes the OpenAPI namespace :).

So I think it would be nice to have a backportable fix (i.e. not a fix included in a version containing a ton of new features).

**Comment by MikeEdgar** [comment]
> So I think it would be nice to have a backportable fix (i.e. not a fix included in a version containing a ton of new features).

Agreed - it would be good to include the fix as a backport in 2.7. It should be as simple as bumping the smallrye-open-api version.
